### PR TITLE
[doc] Pulsar Functions: CLI

### LINKED ISSUE #4554: [docs] Improve the structure of Pulsar Functions
**Is your feature request related to a problem? Please describe.**
I'm a little confused when searching information on Pulsar Functions. The content is scattered in different sections and does not integrate as a whole. Meanwhile, some information duplicates in *Overview* and other sections. It's a little bit frustrating to read the docs as a new user.

**Describe the solution you'd like**
I want to re-structure the architecture of Pulsar Functions as follows, so it will be easier to read and find information.
image

If you have any suggestion on the new architecture, feel free to contact me.
If there is no issue with it, I'll update the content throughout.

### PR DESCRIPTION
**Pull request by Jennifer88huang**
Fixes #4554 

### Motivation
Add a new configure tutorial for new Pulsar Functions structure.

### Modifications
Based on #4846, I'd added the Pulsar Functions configuration content.

### CODE REVIEW

File: site2/docs/functions-configure.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Pulsar Functions configure
```
Comment (sijie) [inline site2/docs/functions-configure.md]: I am not sure if this is a right heading. This page seems to be related to functions command-line tools?

File: site2/docs/functions-configure.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
+# Configure Pulsar Functions
```
Comment (Jennifer88huang) [inline site2/docs/functions-configure.md]:
```suggestion
# Pulsar Functions CLI
```

File: site2/docs/functions-configure.md (hunk 1 above)
Comment (Jennifer88huang) [reply site2/docs/functions-configure.md]: Yes, and we can rename it as "Pulsar Functions CLI", and add another topic "Pulsar Functions config" on how to configure the YAML file.

**Comment by Jennifer88huang** [comment]
@sijie Sure, thanks for your reminder. Done.

**Comment by sijie** [comment]
run java8 tests

**Comment by Jennifer88huang** [comment]
run java8 tests

**Pull request merged**